### ISSUE TITLE
A11y: toggle buttons

### ISSUE THREAD

**Issue by 2LoS** [cppalliance/mrdocs#996]
There are some accessibility issues with the toggle buttons in the documentation pages.
I've prepared a detailed description of these issues, as well as a sample implementation in JS. However, I  prefer to have more competent people to actually edit the scripts :-)

## Visibility

Through the value of which the attribute is set, screen-reader users can also understand whether an element is capable of being expanded or collapsed. In the menu toggles, the `aria-expanded` attribute should be dynamically changed to its opposite value, i.e. from `false` to `true` and vice versa. This part has been successfully achieved.
However, in the documentation pages, menu toggles don't show an outline or another kind of visible change when the keyboard focus is placed on them. This is a problem for those users that exploit keyboard to move throughout the website, but don't use a screen-reader (e.g. individuals with reduced mobility).
To solve this issue, I suggest to keep, at least, the `outline` property of the `<button>` widget.

## Keyboard events

To make the menu toggles accessible for keyboard users, we should support some key events and guarantee that the resulting behavior be as expected. In particular, when the menu is opened, the keyboard focus should be trapped within the menu itself: if the focus is currently on the last item of the menu, and the user tries to move forwards, they should be moved back to the toggle button; conversely, if the is currently on the toggle button, and the user tries to move backwards, they should be moved back to the last item. This guarantees that the user doesn't inadvertently move away from the menu even when it's still expanded. 

In the following list, I've indicated the actions that should be accomplished depending on the key.  
 - **Enter**/**Space**: When the keyboard focus is positioned on the menu toggle, the menu is alternatively displayed and hidden.
 - **Esc**: if the keyboard focus is positioned on one of the menu items, the focus is immediately moved to its toggle button and the menu is closed.
 - **Tab**: When the menu is opened, the tabulation key should move the focus to the next item in the menu. However, if the focus is on the last item, it should go back to the menu toggle.
 -  **Shift** + **Tab**: When the menu is opened, the tabulation and shift keys should move the focus to the previous item in the menu. However, if the focus is on the menu toggle, it should go back to the last item in the list.

## Implementation

To give an idea, I've provided a simple implementation below.

```js
const menuBtn = document.querySelector('#toggle-menu');
const menu = document.querySelector('.navbar');
let active = false;

function toggle() {
	active = !active;
	menuBtn.setAttribute('aria-expanded', active);
}

const list = document.querySelectorAll('a[href]');
const last = list[list.length - 1];

menuBtn.addEventListener('click', toggle);

menu.addEventListener('keydown', (e) => {
	if (active) {
		if (e.key === 'Escape') {
			menuBtn.focus();
			toggle();
		}
	}
});

menuBtn.addEventListener('keydown', (e) => {
	if (active) {
		if (!e.shiftKey && e.key === 'Tab') {
			e.preventDefault();
			menuBtn.focus();
		}
	}
});

last.addEventListener('keydown', (e) => {
	if (active) {
		if (e.shiftKey && e.key === 'Tab') {
			e.preventDefault();
			last.focus();
		}
	}
});
```

**Comment by 2LoS** [comment]
@vinniefalco , it's related to the Mr.Docs website.
